### PR TITLE
Civ 5073 refactor dq model

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-5073

### Change description ###

Refactor existing DirectionQuestionaire model

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts [hunk 1]
```diff
@@ -1,50 +1,33 @@
...
-  considerClaimantDocuments?: ConsiderClaimantDocuments;
-  sharedExpert?: GenericYesNo;
-  requestExtra4weeks?: GenericYesNo;
-  expertCanStillExamine?: ExpertCanStillExamine;
+
   defendantYourselfEvidence?: GenericYesNo;
-  vulnerability?: Vulnerability;
   supportRequired?: SupportRequired;
-  sentExpertReports?: SentExpertReports;
+
```
Comment (portizdev) [inline src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts:12]: extra white line?

Comment (portizdev) [inline src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts:12]: Extra white line?

**Review by commitBlob — APPROVED** [review]

File: src/main/routes/features/directionsQuestionnaire/expertCanStillExamineController.ts [hunk 2]
```diff
@@ -23,7 +25,9 @@ function renderView(form: GenericForm<ExpertCanStillExamine>, res: express.Respo
 expertCanStillExamineController.get(DQ_DEFENDANT_CAN_STILL_EXAMINE_URL, async (req, res, next: express.NextFunction) => {
   try {
     const directionQuestionnaire = await getDirectionQuestionnaire(req.params.id);
-    const expertCanStillExamine = directionQuestionnaire.expertCanStillExamine ? new ExpertCanStillExamine(directionQuestionnaire.expertCanStillExamine.option, directionQuestionnaire.expertCanStillExamine.details) : new ExpertCanStillExamine();
+    const expertCanStillExamine = directionQuestionnaire.experts.expertCanStillExamine ?
+      new ExpertCanStillExamine(directionQuestionnaire.experts.expertCanStillExamine.option,
```
Comment (TatsyMaslova) [inline src/main/routes/features/directionsQuestionnaire/expertCanStillExamineController.ts:29]: you don't need to create a new instance of ExpertCanStillExamine. just assign const to direction.Questionaire.experts.expertCanStillExmine if it exists like this:`const expertCanStillExamine = directionQuestionnaire.experts.expertCanStillExamine ? directionQuestionnaire.experts.expertCanStillExamine : new ExpertCanStillExamine();`

File: src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts [hunk 3]
```diff
@@ -19,7 +21,7 @@ function renderView(form: GenericForm<SentExpertReports>, res: express.Response)
 sentExpertReportsController.get(DQ_SENT_EXPERT_REPORTS_URL, async (req, res, next) => {
   try {
     const directionQuestionnaire = await getDirectionQuestionnaire(req.params.id);
-    const sentExpertReports = directionQuestionnaire.sentExpertReports ? new SentExpertReports(directionQuestionnaire.sentExpertReports.option) : new SentExpertReports();
+    const sentExpertReports = directionQuestionnaire.experts.sentExpertReports ? new SentExpertReports(directionQuestionnaire.experts.sentExpertReports.option) : new SentExpertReports();
```
Comment (TatsyMaslova) [inline src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts:24]: same comment as above - no need to create a new instance from already existing one. only create a new empty form when it does not exist

**Comment by TatsyMaslova** [comment]
As I mentioned before (because i am difficult) the thing i don't like about this approach is that creation of a model is in the controller (I prefer single responsibility because i am simple minded lol), and that we are introducing "any" type (when we tried to get rid of it as much as possible in the beginning of the project). But these are the trade offs and just my opinion and I will go with this approach if everyone thinks its a quicker way to develop this.

File: src/test/unit/services/features/directionsQuestionnaire/vulnerabilityService.test.ts [hunk 4]
```diff
@@ -23,7 +26,8 @@ describe('Vulnerability Service', () => {
     it('should return vulnerability object with isVulnerable no', async () => {
       const claim = new Claim();
       claim.directionQuestionnaire = new DirectionQuestionnaire();
-      claim.directionQuestionnaire.vulnerability = {
+      claim.directionQuestionnaire.vulnerabilityQuestions = new VulnerabilityQuestions();
+      claim.directionQuestionnaire.vulnerabilityQuestions.vulnerability = {
         isVulnerable: 'No',
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/test/unit/services/features/directionsQuestionnaire/vulnerabilityService.test.ts:31]: the scope of the issue is to adapt include the new model structure, this class will be remove in the other refactor issue that I'm working on it. CIV-5074

File: src/test/unit/services/features/directionsQuestionnaire/vulnerabilityService.test.ts [hunk 5]
```diff
@@ -38,7 +42,8 @@ describe('Vulnerability Service', () => {
     it('should return vulnerability object with isVulnerable yes and vulnerabilityDetails', async () => {
       const claim = new Claim();
       claim.directionQuestionnaire = new DirectionQuestionnaire();
-      claim.directionQuestionnaire.vulnerability = {
+      claim.directionQuestionnaire.vulnerabilityQuestions = new VulnerabilityQuestions();
+      claim.directionQuestionnaire.vulnerabilityQuestions.vulnerability = {
         isVulnerable: 'Yes',
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/test/unit/services/features/directionsQuestionnaire/vulnerabilityService.test.ts:47]: the scope of the issue is to adapt include the new model structure, this class will be remove in the other refactor issue that I'm working on it. CIV-5074

File: src/main/routes/features/directionsQuestionnaire/expertCanStillExamineController.ts (hunk 2 above)
Comment (laarmada) [reply src/main/routes/features/directionsQuestionnaire/expertCanStillExamineController.ts:29]: let me fix that in the CIV-5074

File: src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts (hunk 3 above)
Comment (laarmada) [reply src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts:24]: let me fix that in the CIV-5074

**Review by portizdev — APPROVED** [review]

**Comment by portizdev** [comment]
All the code is in 5074, so this PR can be close.

**Pull request closed without merging**